tests/js/server/aql/aql-optimizer-rule-collect-in-cluster.js

```diff
@@ -84,7 +84,7 @@ function optimizerCollectInClusterSuite () {
       });
 
       assertNotEqual(-1, plan.rules.indexOf("collect-in-cluster"));
-      assertEqual(["SingletonNode", "EnumerateCollectionNode", "CollectNode", "RemoteNode", "GatherNode", "CollectNode", "ReturnNode"], nodeTypes);
+      assertEqual(["SingletonNode", "CalculationNode", "EnumerateCollectionNode", "CollectNode", "RemoteNode", "GatherNode", "CollectNode", "ReturnNode"], nodeTypes);
```

Comment:
It looks like the new plan is more expensive than the previous one, as we have an additional CalculationNode now. Can we get rid of this to remove its overhead?

Reply:
Unfortunately this is due to the fact that we are passing null to the COUNT aggregator. This CalculationNode is initializing a temporary variable with null in order to pass it to LENGHT:
```
Execution plan:
 Id   NodeType          Est.   Comment
  1   SingletonNode        1   * ROOT
  3   CalculationNode      1     - LET #2 = null   /* json expression */   /* const assignment */
  6   IndexNode          100     - FOR i IN foo   /* primary index scan, scan only */
  4   CollectNode          1       - COLLECT AGGREGATE cnt = LENGTH(#2)   /* count */
  5   ReturnNode           1       - RETURN cnt
  ```